pkg/apis/provisioning/v1alpha5/provisioner.go

```diff
@@ -70,6 +70,14 @@ type ProvisionerSpec struct {
 	TTLSecondsUntilExpired *int64 `json:"ttlSecondsUntilExpired,omitempty"`
 	// Limits define a set of bounds for provisioning capacity.
 	Limits *Limits `json:"limits,omitempty"`
+	// Consolidation are the consolidation parameters
+	// +optional
+	Consolidation *Consolidation `json:"consolidation,omitempty"`
+}
+
+type Consolidation struct {
+	// Enabled enables consolidation if it has been set
+	Enabled *bool `json:"enabled,omitempty"`
```

Comment:
Booleans in Kubernetes APIs are pretty rare. Things are typically modeled via ptr existence or enums, rather than a big panel of booleans. I'd like to explore configuration alternatives that are more familiar to existing applications in the k8s space.

Reply:
They seem fairly common:

```
find ~/go/pkg/mod/k8s.io/api@v0.24.3 -iname "*.proto" -exec grep "optional bool" {} + | wc -l
     144
```

There is a boolean in Node, Pod, Deployment, ConfigMap, and Secret among others.  This was also the interface in the review design and I don't plan to diverge from the reviewed design in this PR unless there is a clear improvement that can be made.

Reply:
https://github.com/aws/karpenter/pull/2123#issuecomment-1188258308